docs/DYNAMIC_WORKER_PROFILING.md

```diff
@@ -0,0 +1,29 @@
+Dynamic Worker Profiling
```

Comment:
Should we move this doc file to docs/documentation directory?
Also the DYNAMIC_LOG_LEVEL_SETTINGS.md file.